src/lib/cache.js

```diff
@@ -1,24 +1,67 @@
 import { isNull, isArray } from "lodash"
-import { Client } from "memjs"
 import config from "config"
-import { error } from "./loggers"
+import { error, verbose } from "./loggers"
+import redis from "redis"
+import url from "url"
 
-const { NODE_ENV } = process.env
+const { NODE_ENV, OPENREDIS_URL } = process.env
 const { CACHE_LIFETIME_IN_SECONDS } = config
 
-const createClient = () => {
-  if (NODE_ENV === "test") {
-    const store = {}
-    return {
-      store,
-      get: (key, cb) => cb(null, store[key]),
-      set: (key, data) => (store[key] = data),
-    }
+const isTest = NODE_ENV === "test"
+
+const VerboseEvents = ["connect", "ready", "reconnecting", "end", "warning"]
+
+function createMockClient() {
+  const store = {}
+  return {
+    store,
+    get: (key, cb) => cb(null, store[key]),
+    set: (key, data) => (store[key] = data),
+  }
+}
+
+function createRedisClient() {
+  const redisURL = url.parse(OPENREDIS_URL)
+  const client = redis.createClient({
+    host: redisURL.hostname,
+    port: redisURL.port,
+    retryStrategy: options => {
+      if (options.error) {
+        // Errors that lead to the connection being dropped are not emitted to
+        // the error event handler, so send it there ourselves so we can handle
+        // it in one place.
+        // See https://github.com/NodeRedis/node_redis/issues/1202#issuecomment-363116620
+        client.emit("error", error)
+        // End reconnecting on a specific error and flush all commands with a
+        // individual error.
+        if (options.error.code === "ECONNREFUSED") {
+          return new Error("The server refused the connection")
+        }
+      }
+      // End reconnecting after a specific timeout and flush all commands with a
+      // individual error.
+      if (options.total_retry_time > 1000 * 60 * 60) {
```

Comment:
this total retry time seems high... just to clarify the logic here - we're happy to stay in a reconnecting state for up to an hour, but every 10 attempts we flush the command buffer with errors (by returning `undefined` below)?

Reply:
Also unclear on this.  In the constructor, I would opt to fail hard and bail out on the process if we can't establish a connection.

Reply:
Correct. I’m open to suggestions for a better time, this is just the suggested default value from their README.

Reply:
@izakp This is not just for the initial connection but for any situation where the connection was lost.

Reply:
Defaults seem like a good place to start to me (but I was also surprised about the hour).

Reply:
:+1: yeah i'm ok with this, perhaps a little complex for a default strategy. It would be nice to not terminate the process just because the cache is missing (one should be able to run an app slowly without caches). But that is perhaps an idealised view of caching ;)